### ISSUE TITLE
tsc truncates its help output on Mac OS X

### ISSUE THREAD

**Issue by mario-grgic** [Microsoft/TypeScript#8793]
I am on Mac OS X 10.11.5 with node-js v6.2.2 and typescript 1.8.10. Typescript compiler tsc truncates its help message:
```
$ tsc
Version 1.8.10
Syntax:   tsc [options] [file ...]

Examples: tsc hello.ts
          tsc --out file.js file.ts
          tsc @args.txt

Options:
 --allowJs                           Allow javascript files to be compiled.
 --allowSyntheticDefaultImports      Allow default imports from modules with no default export. This does not affect code emit, just typechecking.
 --allowUnreachableCode              Do not report errors on unreachable code.
 --allowUnusedLabels                 Do not report errors on unused labels.
 -d, --declaration                   Generates corresponding '.d.ts' file.
 --experimentalDecorators            Enables experimental support for ES7 decorators.
 --forceConsistentCasingInFileNames  Disallow inconsistently-cased references to the same file.
 -h, --help                          Print this message.
 --init                              Initializes a TypeScript project and creates a tsconfig.json file.
 --jsx KIND                          Specify JSX code generation: 'preserve' or 're$
```

Note that 'react' word got clipped and shell prompt is returned on the same line.

**Comment by DanielRosenwasser** [comment]
@zhengbli or @billti can you try this out and see if you get the same thing (with nightlies and 1.8.10)?

**Comment by Arnavion** [comment]
Someone had hit this a month or so ago on the IRC channel, and as a first test piping the output through `cat` fixed it for them. They were using nvm, so they switched to another node version and then back to the original version, and were unable to reproduce it any more.

**Comment by DanielRosenwasser** [comment]
Sounds potentially like a Mac & Node 6 bug, but we'll need to test it out more on our end as well.

**Comment by blakeembrey** [comment]
I think this is the well-known node 6 issue. There's https://github.com/nodejs/node/issues/6456 tracking it.

**Comment by garthk** [comment]
Given the CTC note on nodejs/node#6456 indicating there'll be no fix coming soon, may I suggest we add the workaround before producing help output?

> CTC Note (Fishrock123): We are working on resolving this. Unfortunately it's not the most trivial thing to investigate or fix well.

**Comment by mhegazy** [comment]
What is the proposed workaround?

**Comment by DanielRosenwasser** [comment]
From the link:

> If this is currently breaking your program, please use this **temporary fix**:

> ```ts
[process.stdout, process.stderr].forEach((s) => {
  s && s.isTTY && s._handle && s._handle.setBlocking &&
    s._handle.setBlocking(true)
})
```

So basically calling `process.stdout._handle.setBlocking(true)` would do it.

But it looks like a flush call when exiting the process would fix it too, and I'd lean towards that. I can't test out a fix because the issue doesn't repro the issue on my Mac.
